### PR TITLE
use a dedicated Dockerfile for alpine variant

### PR DESCRIPTION
**Pull request by ndeloof**
as the master and alpine branches only differ from Dockerfile details, and docker build supports `-f` options for a while, I suggest we remove the `alpine` branch and just use a custom Dockerfile to maintain this image.

### CODE REVIEW

**Comment by carlossg** [comment]
Is there support for -f in official-library or this depend on moving away from it?

**Comment by ndeloof** [comment]
this definitively depends on moving away from it

**Comment by rtyler** [comment]
@ndeloof This pull request doesn't contain any script changes, which I thought would be necessary, to publish the alpine and 'normal' variants to dockerhub

**Comment by ndeloof** [comment]
@rtyler here it is

**Comment by ndeloof** [comment]
my bad, there's something wrong in my script

**Comment by ndeloof** [comment]
@rtyler is it ok to merge and check if we get adequate image pushed on dockerhub as for previous PR ?

**Comment by rtyler** [comment]
The Pipeline run is green but likely unable to report status back just yet due to rate limits, merging.

File: publish.sh [hunk 1]
```diff
@@ -92,7 +79,8 @@ publish() {
     fi
     sha=$(curl -q -fsSL "http://mirrors.jenkins.io/${dir}/${version}/jenkins.war.sha256" | cut -d' ' -f 1)
 
-    docker build --build-arg "JENKINS_VERSION=$version" \
+    docker build --file "Dockerfile$(variant)"
```
Comment (rtyler) [inline publish.sh]: I missed this until it started running, this isn't the right way to do variable substitution in shell:

```
[e_Release_Containers_master-S46EQFXFQYYGMENP66RBTB3BYMGIEHTKSJRTVNBX5BPBEWGZRHXA] Running shell script
+ ./publish.sh
Publishing version: 2.53
./publish.sh: line 82: variant: command not found
"docker build" requires exactly 1 argument(s).
See 'docker build --help'.

Usage:  docker build [OPTIONS] PATH | URL | -

Build an image from a Dockerfile
script returned exit code 1
```

Comment (ndeloof) [inline publish.sh]: Thanks for fixing this @rtyler. All this Jenkinsfile + sh script looks too complex for just publishing a single docker image, isn't it ?

**Pull request merged**